src/logic_utils/PoS_validator.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+#include <cctype>
+#include "PoS_validator.h"
+
+using namespace std;
+
+bool isPoSExpression(const string& expr) {
```

Comment:
I think `isValidPoS` is a better naming. change it here as well as in the headers file.